Program.cs

```diff
@@ -1,4 +1,127 @@
+      /// <summary>Calculate exp part</summary>
+      /// <param name="input">Input number</param>
+      /// <param name="exp">Exponent part</param>
+      /// <returns>Value after exp</returns>
+      static bool AfterExp (string input, out double exp) {
+         exp = 0;
+         double sign;
+         if (input.Contains ('.')) return false;
+         (sign, input) = SignCheck (input);
+         if (!NumericPart (input, out double intPart)) return false;
+         exp = sign * intPart;
+         return true;
+      }
+
+      /// <summary>Calculate value before exp</summary>
+      /// <param name="input">Input number</param>
+      /// <param name="baseNum">Base number</param>
+      /// <returns>Value before exp</returns>
+      static bool BeforeExp (string input, out double baseNum) {
+         baseNum = 0;
+         double sign;
+         (sign, input) = SignCheck (input);
+         string[] parts = input.Split ('.');
+         if (parts.Length > 2) return false;
+         if (!NumericPart (parts[0], out double intPart)) return false;
+         double deciPart = 0;
+         if (parts.Length == 2 && !string.IsNullOrEmpty (parts[1]))
+            if (!DecimalPart (parts[1], out deciPart)) return false;
+         baseNum = sign * (intPart + deciPart);
+         return true;
+      }
+
+      /// <summary>Calculate decimal part</summary>
+      /// <param name="input">Input</param>
+      /// <param name="decPart">Decimal part</param>
+      /// <returns>Value of decimal part</returns>
+      static bool DecimalPart (string input, out double decPart) {
+         decPart = 0;
+         double f = 0.1;
+         foreach (char c in input) {
+            if (!char.IsDigit (c)) return false;
+            decPart += (c - '0') * f;
+            f *= 0.1;
+         }
+         return true;
+      }
+
+      /// <summary>Calculate numeric part</summary>
+      /// <param name="input">Input</param>
+      /// <param name="numPart">Numeric part</param>
+      /// <returns>Value of numeric part</returns>
+      static bool NumericPart (string input, out double numPart) {
+         numPart = 0;
+         foreach (char c in input) {
+            if (!char.IsDigit (c)) return false;
+            numPart = numPart * 10 + (c - '0');
+         }
+         return true;
+      }
+
+      /// <summary>Implementation of double parse</summary>
+      /// <param name="input">Input number</param>
+      /// <param name="res">Double parse value</param>
+      /// <returns>Returns a double precision floating point number whether the input is valid</returns>
+      public static bool Parse (string input, out double res) {
+         res = 0;
+         // Remove the space and change the upper character to lower character
+         input = input.Trim ().ToLower ();
+         if (string.IsNullOrWhiteSpace (input) || input.StartsWith ('e') || input.EndsWith ('e')) return false;
+         if (input.Contains ('e')) {

```

Comment:
You could just use Split and avoid using Contains. 
If input doesnt contain 'e', parts.Length will be just 1